app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -300,7 +310,7 @@ def save_storage_profile_storages_inventory(storage_profile, storages)
     end
 
     save_inventory_multi(storage_profile.storage_profile_storages, hashes,
-                         :use_association, [:storage_profile_id, :storage_id])
+                         [], [:storage_profile_id, :storage_id])
```

Comment:
I dont fully grasp how this `save_inventory_multi` monster works, but, yes, relying on `:dependent => :destroy` should be good :)